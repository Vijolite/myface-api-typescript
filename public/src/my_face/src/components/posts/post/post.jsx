import React, { useState } from "react";
import './post.scss';
import { likePost } from "../../../clients/myFaceClients";

export function Post({ post }) {
//export function Post() {

    // const post = {"id": 130,
    // "message": "Operative system-worthy synergy",
    // "imageUrl": "https://picsum.photos/id/158/600",
    // "createdAt": "2021-01-27 21:55:56",
    // "postedBy": {
    //     "id": 3,
    //     "name": "Valry Gregory",
    //     "username": "vgregory2",
    //     "email": "vgregory2@wsj.com",
    //     "profileImageUrl": "https://robohash.org/vgregory2.png?bgset=bg1",
    //     "coverImageUrl": "https://picsum.photos/id/692/2100/800"
    // }
    // }

    const [whoLikedNames, setWhoLiked]=useState (''); 
    const [whoDislikedNames, setWhoDisliked]=useState ('');  


    function getWhoLiked (arr,names,st_f) {
        let s='';
        if (names==='')
            arr.forEach(person=>s=s+person.name+" ");
        st_f(s);
    }

    return <div className="post">
        <h2>{post.message}</h2>
        <img src={post.imageUrl} alt={post.message} />
        <div>
            <button className="ld_button" onClick={() => likePost(post.id)}>👍</button>
            <p>Posted by {post.postedBy.name} on {post.createdAt}</p>
            <p onClick={function() {getWhoLiked(post.likedBy,whoLikedNames,setWhoLiked)}}> 
            Liked by {post.likedBy.length}: {whoLikedNames}</p>
            <p onClick={function() {getWhoLiked(post.dislikedBy,whoDislikedNames,setWhoDisliked)}}> 
            Disliked by {post.dislikedBy.length}: {whoDislikedNames}</p>
        </div>
    </div>
}

