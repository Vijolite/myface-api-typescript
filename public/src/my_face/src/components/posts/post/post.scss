@import "../../../constants.scss";


.ld_button {
    position: absolute;
    top:15px;
    right: 15px;
    width: 30px;
    height: 30px;
}
//background-color: $backgroundColor;

.post {
    background-color: $insideBackgroundColor;//$backgroundColor;
    color: $textColor;
    display: grid;
    grid-template-columns: 0.4fr 0.6fr;//40vw 1fr;
    grid-template-rows: 0.6fr 0.4fr;
    height: 40vw; 
    margin: 10px;
    border-color: $borderColor;
    border-style: solid;
    img {
        grid-column: 1/2;
        grid-row: 1/3;
        // width: 40vw;
        // height: 40vw;
        width: 100%;
        height: 100%;
    }
    h2 {
        grid-column: 2/3;
        grid-row: 1/2;
        align-self: center;
        font-size: 1em;
    }
    div {
        grid-column: 2/3;
        grid-row: 2/3;
        
        p {
            font-size: 0.5rem;
            margin:0;
        }
    }
    
}


@media (min-width: 600px) {
    .post {
        height: 150px;
        img {
            height: 150px;
        }
    }
}
// @media (min-width: 900px) {
//     .post_item {
//         width: calc(95%/3);
//     }
// }