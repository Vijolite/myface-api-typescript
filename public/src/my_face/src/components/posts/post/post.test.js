import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Post } from "./post.jsx";


test("renders message", () => {
  // Arrange
  const post = {
    message: "Post message",
    imageUrl: "Image url",
    createdAt: "2022/04/05 11:00",
    postedBy: {
      name: "User name",
    },
    likedBy: {
      length:"10",
    },
    dislikedBy: {
      length:"5",
    },
  };

  // Act
  render(<Post post={post} />);
  const messageElement = screen.getByText(/post message/i);

  // Assert
  expect(messageElement).toBeInTheDocument();
});

test("matches snapshot", () => {
  // Arrange
  const post = {
    message: "Post message",
    imageUrl: "Image url",
    createdAt: "2022/04/05 11:00",
    postedBy: {
      name: "User name",
    },
    likedBy: {
      length:"10",
    },
    dislikedBy: {
      length:"5",
    },
  };

  // Act
  const tree = renderer
    .create(<Post post={post} />)
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});

