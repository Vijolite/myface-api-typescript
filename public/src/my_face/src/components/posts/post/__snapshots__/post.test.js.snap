// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="post"
>
  <h2>
    Post message
  </h2>
  <img
    alt="Post message"
    src="Image url"
  />
  <div>
    <p>
      Posted by 
      User name
       on 
      2022/04/05 11:00
    </p>
    <p
      onClick={[Function]}
    >
      Liked by 
      10
      : 
      
    </p>
    <p
      onClick={[Function]}
    >
      Disliked by 
      5
      : 
      
    </p>
  </div>
</div>
`;
